### PR TITLE
i54 Get UI working with Dual Boot Conditionals

### PR DESCRIPTION
**Pull request by ShanaLMoore**
This PR corrects issues that prevented a dev from seeing the UI/homepage when dual booting. This was mainly done by adding dual boot conditionals throughout the code. From here devs should be unblocked from subsequent tickets.

ref: 
- https://github.com/scientist-softserv/ams/issues/54

## NORMAL boot: 
image

## Booting Rails 6.1: 
image

  **- Note: I reverted a commit that allowed the user to navigate to the dashboard without error** This will get resolved in a separate [PR](https://github.com/WGBH-MLA/ams/pull/745).

### CODE REVIEW

File: app/helpers/solr_helper.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# app/helpers/solr_helper.rb
+module SolrHelper
+  def solr_name(base_name)
+    if Module.const_defined?(:Solrizer)
```
Comment (ShanaLMoore) [inline app/helpers/solr_helper.rb]: This is a more valid questions instead of asking if App.rails_5_1?

**Comment by ShanaLMoore** [comment]
converting back to draft. I expected the test part of the pipeline to be green. Diving in!

File: app/helpers/solr_helper.rb (hunk 1 above)
Comment (ShanaLMoore) [reply app/helpers/solr_helper.rb]: moved to a separate PR

**Review by orangewolf — APPROVED** [review]

File: app/views/catalog/_search_form.html.erb [hunk 2]
```diff
@@ -16,7 +16,11 @@
     <% end %>
 
     <label for="q" class="sr-only"><%= t('blacklight.search.form.search.label') %></label>
-    <%= text_field_tag :q, params[:q], placeholder: t('blacklight.search.form.search.placeholder'), class: "search_q q form-control", id: "q", autofocus: should_autofocus_on_search_box?, data: { autocomplete_enabled: autocomplete_enabled?, autocomplete_path: blacklight.suggest_index_path }  %>
+    <% if App.rails_5_1? %>
+      <%= text_field_tag :q, params[:q], placeholder: t('blacklight.search.form.search.placeholder'), class: "search_q q form-control", id: "q", autofocus: should_autofocus_on_search_box?, data: { autocomplete_enabled: autocomplete_enabled?, autocomplete_path: blacklight.suggest_index_path }  %>
+    <% else %>
+      <%= text_field_tag :q, params[:q], placeholder: t('blacklight.search.form.search.placeholder'), class: "search_q q form-control", id: "q", autofocus: should_autofocus_on_search_box?, data: { autocomplete_enabled: autocomplete_enabled? }  %>
```
Comment (kirkkwang) [inline app/views/catalog/_search_form.html.erb:22]: Seems like we can remove the entire `data` argument here, found this commit fwiw

**Removed the data config item to fix error: undefined method `suggest_index_path'**
https://gitlab.galileo.usg.edu/DLG/meta/-/merge_requests/359/diffs?commit_id=1157838179ef6d22946a33dc1329ebfe52d6afa5

Comment (ShanaLMoore) [reply app/views/catalog/_search_form.html.erb:22]: Good find! But since I already merged this PR let's make another one.

**Pull request merged**